### ISSUE TITLE
Please update BlackBoxOptim to use Distributions.jl 0.24.15

### ISSUE THREAD

**Issue by StevenSiew** [robertfeldt/BlackBoxOptim.jl#171]
Please update BlackBoxOptim to use Distributions.jl 0.24.15

I think BlackBoxOptim is insisting that it need Distributions.jl 0.23.11

**Comment by robertfeldt** [comment]
I added debs based on CompatHelper bot suggestions. Please check on your side and report back if any problems. Install from master for now.

**Comment by StevenSiew** [comment]
Thanks, it seems to work when i tested Master on it.

`(@v1.5) pkg> activate testdist
 Activating new environment at `~/juliascript/testdist/Project.toml`

(testdist) pkg> st
Status `~/juliascript/testdist/Project.toml` (empty project)

(testdist) pkg> add https://github.com/robertfeldt/BlackBoxOptim.jl
    Cloning git-repo `https://github.com/robertfeldt/BlackBoxOptim.jl`
   Updating git-repo `https://github.com/robertfeldt/BlackBoxOptim.jl`
   Updating registry at `~/.julia/registries/General`
   Updating git-repo `https://github.com/JuliaRegistries/General.git`
  Resolving package versions...
  Installed PDMats ─────────── v0.11.0
  Installed FillArrays ─────── v0.11.7
  Installed SpecialFunctions ─ v1.3.0
  Installed Distributions ──── v0.24.15
Updating `~/juliascript/testdist/Project.toml`
  [a134a8b2] + BlackBoxOptim v0.5.0 `https://github.com/robertfeldt/BlackBoxOptim.jl#master`
Updating `~/juliascript/testdist/Manifest.toml`
  [56f22d72] + Artifacts v1.3.0
  [a134a8b2] + BlackBoxOptim v0.5.0 `https://github.com/robertfeldt/BlackBoxOptim.jl#master`
  [a9c8d775] + CPUTime v1.0.0
  [d360d2e6] + ChainRulesCore v0.9.29
  [34da2185] + Compat v3.25.0
  [e66e0078] + CompilerSupportLibraries_jll v0.3.4+0
  [9a962f9c] + DataAPI v1.6.0
  [864edb3b] + DataStructures v0.18.9
  [31c24e10] + Distributions v0.24.15
  [1a297f60] + FillArrays v0.11.7
  [cd3eb016] + HTTP v0.9.5
  [83e8ac13] + IniFile v0.5.0
  [692b3bcd] + JLLWrappers v1.2.0
  [682c06a0] + JSON v0.21.1
  [739be429] + MbedTLS v1.0.3
  [c8ffd9c3] + MbedTLS_jll v2.16.8+1
  [e1d29d7a] + Missings v0.4.5
  [ca575930] + NetworkOptions v1.2.0
  [efe28fd5] + OpenSpecFun_jll v0.5.3+4
  [bac558e1] + OrderedCollections v1.4.0
  [90014a1f] + PDMats v0.11.0
  [69de0a69] + Parsers v1.1.0
  [1fd47b50] + QuadGK v2.4.1
  [79098fc4] + Rmath v0.6.1
  [f50d1b31] + Rmath_jll v0.2.2+2
  [a2af1166] + SortingAlgorithms v0.3.1
  [d4ead438] + SpatialIndexing v0.1.2
  [276daf66] + SpecialFunctions v1.3.0
  [2913bbd2] + StatsBase v0.33.4
  [4c63d2b9] + StatsFuns v0.9.6
  [5c2747f8] + URIs v1.2.0
  [2a0f44e3] + Base64
  [ade2ca70] + Dates
  [8bb1440f] + DelimitedFiles
  [8ba89e20] + Distributed
  [b77e0a4c] + InteractiveUtils
  [76f85450] + LibGit2
  [8f399da3] + Libdl
  [37e2e46d] + LinearAlgebra
  [56ddb016] + Logging
  [d6f4376e] + Markdown
  [a63ad114] + Mmap
  [44cfe95a] + Pkg
  [de0858da] + Printf
  [3fa0cd96] + REPL
  [9a3f8284] + Random
  [ea8e919c] + SHA
  [9e88b42a] + Serialization
  [1a1011a3] + SharedArrays
  [6462fe0b] + Sockets
  [2f01184e] + SparseArrays
  [10745b16] + Statistics
  [4607b0f0] + SuiteSparse
  [8dfed614] + Test
  [cf7118a7] + UUIDs
  [4ec0a83e] + Unicode


(testdist) pkg> add Distributions
  Resolving package versions...
Updating `~/juliascript/testdist/Project.toml`
  [31c24e10] + Distributions v0.24.15
No Changes to `~/juliascript/testdist/Manifest.toml`


`

**Comment by Alexander-Barth** [comment]
I run the test suite on commit 83c811671c00fffdcaadf042ad07ee2e2194ecbf and all seems fine (see below).
Thank you for making this package. It was very [helpful for me](https://link.springer.com/article/10.1007/s10236-020-01432-x).
Did you see this announcement concerning [Travis CI](https://discourse.julialang.org/t/easy-workflow-file-for-setting-up-github-actions-ci-for-your-julia-package/49765)? If you want to change to Github Action, but have any issue feel free to ping me.


```
(@v1.5) pkg> dev BlackBoxOptim
    Cloning git-repo `https://github.com/robertfeldt/BlackBoxOptim.jl.git`
  Resolving package versions...
  Installed Distributions ─ v0.24.15
Updating `~/.julia/environments/v1.5/Project.toml`
  [a134a8b2] ~ BlackBoxOptim v0.5.0 ⇒ v0.5.0 `~/.julia/dev/BlackBoxOptim`
  [31c24e10] ↑ Distributions v0.23.11 ⇒ v0.24.15
  [276daf66] ↑ SpecialFunctions v0.10.3 ⇒ v1.3.0
Updating `~/.julia/environments/v1.5/Manifest.toml`
  [4c555306] ↑ ArrayLayouts v0.4.11 ⇒ v0.4.12
  [a134a8b2] ~ BlackBoxOptim v0.5.0 ⇒ v0.5.0 `~/.julia/dev/BlackBoxOptim`
  [31c24e10] ↑ Distributions v0.23.11 ⇒ v0.24.15
  [1a297f60] ↑ FillArrays v0.9.7 ⇒ v0.10.2
  [276daf66] ↑ SpecialFunctions v0.10.3 ⇒ v1.3.0

(@v1.5) pkg> test BlackBoxOptim
    Testing BlackBoxOptim
Status `/tmp/jl_v6Ltdv/Project.toml`
  [a134a8b2] BlackBoxOptim v0.5.0 `~/.julia/dev/BlackBoxOptim`
  [a9c8d775] CPUTime v1.0.0
  [336ed68f] CSV v0.8.4
  [34da2185] Compat v3.25.0
  [31c24e10] Distributions v0.24.15
  [cd3eb016] HTTP v0.9.5
  [682c06a0] JSON v0.21.1
  [d4ead438] SpatialIndexing v0.1.2
  [2913bbd2] StatsBase v0.33.4
  [8ba89e20] Distributed
  [37e2e46d] LinearAlgebra
  [de0858da] Printf
  [9a3f8284] Random
  [ea8e919c] SHA
  [9e88b42a] Serialization
  [8dfed614] Test
Status `/tmp/jl_v6Ltdv/Manifest.toml`
  [56f22d72] Artifacts v1.3.0
  [a134a8b2] BlackBoxOptim v0.5.0 `~/.julia/dev/BlackBoxOptim`
  [a9c8d775] CPUTime v1.0.0
  [336ed68f] CSV v0.8.4
  [d360d2e6] ChainRulesCore v0.9.29
  [34da2185] Compat v3.25.0
  [e66e0078] CompilerSupportLibraries_jll v0.3.4+0
  [9a962f9c] DataAPI v1.6.0
  [864edb3b] DataStructures v0.18.9
  [e2d170a0] DataValueInterfaces v1.0.0
  [31c24e10] Distributions v0.24.15
  [1a297f60] FillArrays v0.10.2
  [cd3eb016] HTTP v0.9.5
  [83e8ac13] IniFile v0.5.0
  [82899510] IteratorInterfaceExtensions v1.0.0
  [692b3bcd] JLLWrappers v1.2.0
  [682c06a0] JSON v0.21.1
  [739be429] MbedTLS v1.0.3
  [c8ffd9c3] MbedTLS_jll v2.16.8+1
  [e1d29d7a] Missings v0.4.5
  [ca575930] NetworkOptions v1.2.0
  [efe28fd5] OpenSpecFun_jll v0.5.3+4
  [bac558e1] OrderedCollections v1.4.0
  [90014a1f] PDMats v0.10.1
  [69de0a69] Parsers v1.1.0
  [2dfb63ee] PooledArrays v1.2.1
  [1fd47b50] QuadGK v2.4.1
  [79098fc4] Rmath v0.6.1
  [f50d1b31] Rmath_jll v0.2.2+2
  [91c51154] SentinelArrays v1.2.16
  [a2af1166] SortingAlgorithms v0.3.1
  [d4ead438] SpatialIndexing v0.1.2
  [276daf66] SpecialFunctions v1.3.0
  [2913bbd2] StatsBase v0.33.4
  [4c63d2b9] StatsFuns v0.9.6
  [3783bdb8] TableTraits v1.0.0
  [bd369af6] Tables v1.4.1
  [5c2747f8] URIs v1.2.0
  [2a0f44e3] Base64
  [ade2ca70] Dates
  [8bb1440f] DelimitedFiles
  [8ba89e20] Distributed
  [9fa8497b] Future
  [b77e0a4c] InteractiveUtils
  [76f85450] LibGit2
  [8f399da3] Libdl
  [37e2e46d] LinearAlgebra
  [56ddb016] Logging
  [d6f4376e] Markdown
  [a63ad114] Mmap
  [44cfe95a] Pkg
  [de0858da] Printf
  [3fa0cd96] REPL
  [9a3f8284] Random
  [ea8e919c] SHA
  [9e88b42a] Serialization
  [1a1011a3] SharedArrays
  [6462fe0b] Sockets
  [2f01184e] SparseArrays
  [10745b16] Statistics
  [4607b0f0] SuiteSparse
  [8dfed614] Test
  [cf7118a7] UUIDs
  [4ec0a83e] Unicode

.....┌ Warning: MultithreadEvaluator tests require >1 threads, 1 found, use JULIA_NUM_THREADS
└ @ Main.BlackBoxOptimTests ~/.julia/dev/BlackBoxOptim/test/test_evaluator.jl:110
................┌ Warning: BorgMOEA MultithreadEvaluator tests require >1 threads, 1 found, use JULIA_NUM_THREADS
└ @ Main.BlackBoxOptimTests ~/.julia/dev/BlackBoxOptim/test/test_borg_moea.jl:21
..┌ Warning: Top-level MultithreadEvaluator tests require >1 threads, 1 found, use JULIA_NUM_THREADS
└ @ Main.BlackBoxOptimTests ~/.julia/dev/BlackBoxOptim/test/test_toplevel_bboptimize.jl:90
┌ Warning: Optimization interrupted, recovering intermediate results...
└ @ BlackBoxOptim ~/.julia/dev/BlackBoxOptim/src/opt_controller.jl:471
Starting optimization with optimizer BlackBoxOptim.DiffEvoOpt{BlackBoxOptim.FitPopulation{Float64},BlackBoxOptim.RadiusLimitedSelector,BlackBoxOptim.AdaptiveDiffEvoRandBin{3},BlackBoxOptim.RandomBound{BlackBoxOptim.ContinuousRectSearchSpace}}
0.00 secs, 0 evals, 0 steps

Optimization stopped after 10001 steps and 0.02 seconds
Termination reason: Max number of steps (10000) reached
Steps per second = 425070.52
Function evals per second = 429193.29
Improvements/step = 0.34030
Total function evaluations = 10098


Best candidate found: [-4.92131e-19, -4.69709e-19]

Fitness: 0.000000000

......
Test Summary:            |  Pass  Broken  Total
BlackBoxOptim test suite | 40853       3  40856
Tested 29 files in 188.5 seconds.
    Testing BlackBoxOptim tests passed 

(@v1.5) pkg> 
```

**Comment by robertfeldt** [comment]
Thanks to you both. Seems ok for now. I plan to run some more tests and minor updates on master and then release a new release so people don't need to be on master.

And great to hear it is being used in research. This package started out as something we used in my research group (https://scholar.google.se/citations?hl=sv&user=VemqkeEAAAAJ) and there will be more updates coming soon again when we have some more related projects.

Closing this.
